vertx-sql-client/src/main/java/io/vertx/sqlclient/SqlClient.java

```diff
@@ -36,136 +30,21 @@
-   * @param handler the handler notified with the execution result
-   * @return a reference to this, so the API can be used fluently
-   */
-  @Fluent
-  SqlClient query(String sql, Handler<AsyncResult<RowSet<Row>>> handler);
-
-  /**
-   * Like {@link #query(String, Handler)} but returns a {@code Future} of the asynchronous result
-   */
-  Future<RowSet<Row>> query(String sql);
-
-  /**
-   * Execute a simple query using the given {@code sql} string, the asynchronous result is represented as a collection of elements transformed by the provided {@link java.util.stream.Collector}.
-   *
-   * @param sql the query SQL
-   * @param collector the collector
-   * @param handler the handler notified with the execution result
-   * @return a reference to this, so the API can be used fluently
-   */
-  @GenIgnore
-  <R> SqlClient query(String sql, Collector<Row, ?, R> collector, Handler<AsyncResult<SqlResult<R>>> handler);
-
-  /**
-   * Like {@link #query(String, Collector, Handler)} but returns a {@code Future} of the asynchronous result
-   */
-  @GenIgnore
-  <R> Future<SqlResult<R>> query(String sql, Collector<Row, ?, R> collector);
-
-  /**
-   * Execute the given {@code sql} string using a prepared statement without any parameter, the asynchronous result is represented as a {@link RowSet}.
-   *
-   * @param sql the prepared query SQL
-   * @param handler the handler notified with the execution result
-   * @return a reference to this, so the API can be used fluently
-   */
-  @Fluent
-  SqlClient preparedQuery(String sql, Handler<AsyncResult<RowSet<Row>>> handler);
-
-  /**
-   * Like {@link #preparedQuery(String, Handler)} but returns a {@code Future} of the asynchronous result
-   */
-  Future<RowSet<Row>> preparedQuery(String sql);
-
-  /**
-   * Execute the given {@code sql} string using a prepared statement without any parameter, the asynchronous result is represented as a collection of elements transformed by the provided {@link java.util.stream.Collector}.
+   * Create a query.
    *
-   * @param sql the prepared query SQL
-   * @param collector the collector
-   * @param handler the handler notified with the execution result
-   * @return a reference to this, so the API can be used fluently
-   */
-  @GenIgnore
-  <R> SqlClient preparedQuery(String sql, Collector<Row, ?, R> collector, Handler<AsyncResult<SqlResult<R>>> handler);
-
-  /**
-   * Like {@link #preparedQuery(String, Collector, Handler)} but returns a {@code Future} of the asynchronous result
+   * @return the query
    */
-  @GenIgnore
-  <R> Future<SqlResult<R>> preparedQuery(String sql, Collector<Row, ?, R> collector);
+  Query<RowSet<Row>> query(String sql);
 
   /**
-   * Execute the given {@code sql} string using a prepared statement with parameters set in the {@code Tuple}, the asynchronous result is represented as a {@link RowSet}.
+   * Create a prepared query, one of the {@link PreparedQuery#execute}, {@link PreparedQuery#executeBatch}
```

Comment:
the created object is actually not a **prepared** query, how about writing as 
> Create a prepared query request, ...

Reply:
I'm not sure to understand here

Reply:
I mean we could clarify in the documentation that this created `PreparedQuery` object is not prepared indeed and is only a request handle for one-shot executing a prepared query.

Reply:
I think it does not matter because it is the same from the user perspective as he will get a single callback that encompass both query validation and query execution which is what makes the difference and the doc says that `execute` must be called to execute the query